Add announcement for Stats Revamp

guarani opened the pull request:
Addresses https://github.com/wordpress-mobile/WordPress-iOS/issues/18624

This PR adds the UI for the Stats Revamp's feature announcement. It doesn't implement the functionality for showing the announcement, that will be done in a later PR.

To test:

1. In Xcode, open the app delegate and temporarily add a call to `showStatsRevampV2FeatureIntroduction()` to the end of the `applicationDidBecomeActive` method, like so: `WPTabBarController.sharedInstance().showStatsRevampV2FeatureIntroduction()`.
2. Open the app, tap Profile pic → App Settings → Debug Settings → turn "New Cards for Stats Insights" ON
3. Background the app
4. Foreground the app 
5. Notice that the feature announcement is shown
6. Compare the screen to the design shown in Figma (the one shown in the linked issue may be outdated)
    - The image that's shown on the announcement screen appears to have a faint light blue background. When I exported the image from Figma as a PDF for inclusion in the asset catalogue, its background appears to be light gray. I've left it as-is as I'm not sure what's going on and the different might be only very slight or unnoticeable. Let me know if you have any pointers here, otherwise I'm happy to leave it as-is.

## Regression Notes
1. Potential unintended areas of impact

This PR doesn't implement functionally for showing the announcement, so it's unlikely this can negatively impact other parts of the app.

8. What I did to test those areas of impact (or what existing automated tests I relied on)

Not applicable.

9. What automated tests I added (or what prevented me from doing so)

This PR is just UI-related, without functionality, so I haven't looked at adding tests. In a follow-up PR when the screen's functionality is implemented, I can reconsider tests. Let me know if you think I should add tests here.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Commit:
Add announcement for Stats Revamp

frosty commented on WordPress/Classes/ViewRelated/Feature Introduction/Stats Revamp v2/StatsRevampV2IntroductionPresenter.swift:
```diff
@@ -0,0 +1,32 @@
+    }()
+
+    private lazy var selectedBlog: Blog? = {
+        return AccountService(managedObjectContext: ContextManager.shared.mainContext).defaultWordPressComAccount()?.visibleBlogs.first
```
I wonder if we might want to use BlogService's `lastUsedOrFirstBlog` or `lastUsedOrPrimaryBlog`?

guarani replied:
Good idea! I hadn't seen those helper methods. I then realized that this line in question is not used for now, since this PR doesn't include actual functionality for the screen. I'll go ahead and remove it for now, but will keep these methods in mind for next time 👍

frosty commented on WordPress/Classes/ViewRelated/Feature Introduction/Stats Revamp v2/StatsRevampV2FeatureDescriptionView.swift:
```diff
@@ -0,0 +1,88 @@
+    enum Strings {
+        static let featureDescription: String = NSLocalizedString("Insights help you understand how your content resonates with your audience with an overview of how it is performing and guidance to your next steps.", comment: "Description of updated Stats Insights displayed in the Feature Introduction view.")
+        static let noteLabel: String = NSLocalizedString("Note:", comment: "Label for the note displayed in the Feature Introduction view.")
+        static let noteText: String = NSLocalizedString("You can learn more about Insights at an time in My Site > Stats > Insights", comment: "Note displayed in the Feature Introduction view for the updated Stats Insights feature.")
```
Slight typo: "at an time" -> "at any time"

guarani replied:
Thanks for spotting that!

frosty reviewed (commented):
Looks good! I tried re-exporting the image, and it's better but I think something is slightly off with the aspect ratio as you can see the background or shadow showing at the top and bottom:

<img width="372" alt="image" src="https://user-images.githubusercontent.com/4780/171187160-e73d4c8d-3cf4-4fe6-94b1-e5e4105bc116.png">

Other than that, I just spotted one small typo and one suggestion for a code tweak.

Here's the updated PDF:

[stats-revamp-v2-feature-intro-preview.pdf](https://github.com/wordpress-mobile/WordPress-iOS/files/8805620/stats-revamp-v2-feature-intro-preview.pdf)

Commit:
Fix typo in Stats Revamp feature announcement

Commit:
Removed unused code

Commit:
Fixed Stats Revamp feature announcement image

Commit:
Add Stats Revamp announcement header img

Commit:
Fixed Stats Revamp annoucement layout warnings

guarani commented:
Hi @frosty, this is ready for another review.

Thanks for the exported asset PDF, I'm not sure why it didn't export properly for me. I went ahead and used the file you attached here.

One thing I noticed was that the Feature Announcement screen that was developed for Blogging Prompts shows a "header image"; however, the designs don't include this. I went ahead and added it in 41e440739214a16b4114410015f0aa7ab289c6eb. This is what it looks like now (just let me know if you think it shouldn't be there):

| <img src="https://user-images.githubusercontent.com/1898325/171313352-488239de-27e5-48fa-b714-98fcb43343b5.png" alt="Screenshot showing new light bulb shaped header image" width="350"> |
| - |

Another couple of things: 
- I had duplicate copy (both in code and in the xib), so I left the in-code copy and removed the one in the xib.
- I fixed a ambiguous layout warning in the xib that I hadn't noticed earlier.

🙇

frosty reviewed (approved):
Changes look good! The only remaining issue I'm seeing is the shadow under / around the main image, and I can't quite work out why it's there:

<img width="354" alt="Screenshot 2022-06-02 at 11 04 15" src="https://user-images.githubusercontent.com/4780/171612350-c9b0e73d-6695-40f7-82f7-f05f7eabe190.png">

This isn't in the designs, and I think it looks a little off tbh – but totally something we can address separately if necessary:

<img width="387" alt="Screenshot 2022-06-02 at 11 31 30" src="https://user-images.githubusercontent.com/4780/171612429-0cb63e75-31c0-4336-a668-51dabb5c705a.png">

guarani commented:
> The only remaining issue I'm seeing is the shadow under / around the main image, and I can't quite work out why it's there:

That's strange, while I can see the shadow in the (first) image, I don't see it on the device nor on the image asset:

| <img width="1009" alt="Screen Shot 2022-06-02 at 15 01 18" src="https://user-images.githubusercontent.com/1898325/171717194-d7566dad-e89d-4bf6-a540-757c82f49935.png"> |
| - |

For reference, this is the `.pdf` file from Xcode (which I took from [your above comment](https://github.com/wordpress-mobile/WordPress-iOS/pull/18738#pullrequestreview-990461226)), and I can't see the shadow their either:

[stats-revamp-v2-feature-intro-preview.pdf](https://github.com/wordpress-mobile/WordPress-iOS/files/8826828/stats-revamp-v2-feature-intro-preview.pdf)

---

To clarify, when I mentioned earlier [at the start of this PR](https://github.com/wordpress-mobile/WordPress-iOS/pull/18738#issue-1248904825) that the image had the wrong background, I'd exported the layer named `iOS-imgAsset 1`:

| <img width="275" alt="Screen Shot 2022-06-02 at 15 15 14" src="https://user-images.githubusercontent.com/1898325/171719921-fff043b8-bfb8-464e-9318-2040f88e876c.png"> |
| - |

I then realized that has a transparent background, but had just downloaded the PDF you provided earlier. Today, I tried exporting the parent layer `Group 25151` and it includes the correct light-blue background color:

<img width="309" alt="Screen Shot 2022-06-02 at 15 17 11" src="https://user-images.githubusercontent.com/1898325/171720172-849dd043-a08f-43d9-9328-fb4f83fae14a.png">

I presume the latter export is the correct one, and is the same one you exported, but please let me know if I'm exporting incorrectly. Either way, I can't seem to reproduce the issue of the shadow appearing under the image.

--- 

To help me fix this, could you please tell me which device (or sim) you were using, and send a full screenshot showing the shadow? 🙇

frosty commented:
It turns out the shadow is just due to the way the `StatsRevampV2FeatureDescriptionView` is constructed:

```
cardImageView.layer.shadowOffset = Style.cardShadowOffset
cardImageView.layer.shadowOpacity = Style.cardShadowOpacity
cardImageView.layer.shadowRadius = Style.cardShadowRadius
```

It's intentional in the code, but I don't know why you're not seeing it. Based on the designs though, I think we should remove this.

| With shadow | Without |
|---|---|
| ![Simulator Screen Shot - iPhone 13 Pro - 2022-06-07 at 11 52 24](https://user-images.githubusercontent.com/4780/172362615-a8b7e0be-a573-4dd4-808d-2f46adcd1d06.png) | ![Simulator Screen Shot - iPhone 13 Pro - 2022-06-07 at 11 28 50](https://user-images.githubusercontent.com/4780/172362578-18678f75-d024-4f5c-89e7-f0a3c8d82050.png) |

guarani commented:
Ah, not sure why I don't see it nor how I didn't associate that code – which I copied across – to this.
Thanks, will change this as soon as I can.

mokagio commented:
Just a note to say that I considered merging this PR because it's part of the 20.1 milestone and I'm about to start that code freeze.

I did not do it because of the merge conflicts. 

Normally, I'd just bump the milestone, but this is an "announcement PR", so I think it ought to be part of 20.1. Right?

I'll ship a new beta with this as soon as it merges.

mokagio commented:
> Normally, I'd just bump the milestone, but this is an "announcement PR", so I think it ought to be part of 20.1. Right?

Under that assumption, I updated this PR's base branch to `release/20.1`. Feel free to revert to `trunk` if my assumption turns out to be incorrect.

frosty commented:
@mokagio It's fine to not merge this, but thanks for considering it. We only need this in once we've completed the feature, and it's not ready to go yet :)

guarani commented:
Thanks for looking at this folks! I should have bumped this to the `20.2` milestone since [there's an updated needed from my side](https://github.com/wordpress-mobile/WordPress-iOS/pull/18738#issuecomment-1149257910), and the conflict resolution mentioned by @mokagio, that are needed before merging. Sorry for the confusion folks!

I've updated the PR's base branch to `trunk` and bumped the milestone to 20.2.

🙇

guarani commented:
Thanks for looking at this folks! I should have bumped this to the `20.2` milestone since [there's an updated needed from my side](https://github.com/wordpress-mobile/WordPress-iOS/pull/18738#issuecomment-1149257910), and the conflict resolution mentioned by @mokagio, that are needed before merging. Sorry for the confusion folks!

I've updated the PR's base branch to `trunk` and bumped the milestone to 20.2.

🙇

guarani commented:
Thanks for looking at this folks! I should have bumped this to the `20.2` milestone since [there's an updated needed from my side](https://github.com/wordpress-mobile/WordPress-iOS/pull/18738#issuecomment-1149257910), and the conflict resolution mentioned by @mokagio, that are needed before merging. Sorry for the confusion folks!

I've updated the PR's base branch to `trunk` and bumped the milestone to 20.2.

🙇

mokagio commented:
@guarani @frosty no worries, and thank you for the context 👍

Commit:
Merge branch 'trunk' into feature/stats-revamp-announcement

Commit:
Remove shadow from Stats Revamp announcement

guarani commented:
Hi @frosty, I removed the unwanted shadow. Here's the before/after for this change:

| Before | After |
| - | - |
| <img width="542" alt="Screen Shot 2022-06-20 at 14 32 46" src="https://user-images.githubusercontent.com/1898325/174661368-3c41ee88-3608-47b0-b862-e9b1f5b644a6.png"> | <img width="542" alt="Screen Shot 2022-06-20 at 14 35 48" src="https://user-images.githubusercontent.com/1898325/174661351-4c323981-ae87-48c3-84cc-5fe1b0795415.png"> |

On a side note: In the above "before" screenshot, the shadow is to the sides (not above nor below) the image. I'm not sure why the shadow in your "with shadow" [screenshot above](https://github.com/wordpress-mobile/WordPress-iOS/pull/18738#issuecomment-1148508388) is instead above and below the image. My guess is that the code that adds the shadow was getting called before the layout had completed. Either way, it's not super important now since the shadow has now been removed altogether, but it might explain why I didn't see it earlier (e.g. if I was looking for a shadow in the wrong place).

guarani commented:
I've set this to auto-merge since there haven't been significant changes since your approval @frosty. Also, a follow-up PR will add the logic for displaying the feature announcement at the right time.
